### PR TITLE
GODRIVER-2156 Enable unparam linter.

### PR DESCRIPTION
**Pull request by matthewdale**
Enable the `unparam` linter and remove unused function parameters.

### CODE REVIEW

**Review by benjirewis — APPROVED** [review]
LGTM thanks for all the additional cleanup beyond just removing the `unused` parameters.

**Review by kevinAlbs — APPROVED** [review]
LGTM with a suggested addition for the `MinTLSVersion` change as part of a separate PR.

File: mongo/options/clientoptions.go [hunk 1]
```diff
@@ -718,6 +718,9 @@ func (c *ClientOptions) SetSocketTimeout(d time.Duration) *ClientOptions {
 //
 // The default is nil, meaning no TLS will be enabled.
 func (c *ClientOptions) SetTLSConfig(cfg *tls.Config) *ClientOptions {
+	if cfg.MinVersion == 0 {
+		cfg.MinVersion = tls.VersionTLS12
+	}
```
[the remark this answers is not in the document]
Comment (matthewdale) [reply mongo/options/clientoptions.go:723]: Good catch! That actually isn't intended to be part of this PR. Will remove.

[the remark this answers is not in the document]
Comment (matthewdale) [reply mongo/options/clientoptions.go]: Removed.

**Comment by matthewdale** [comment]
One failing test is due to [DRIVERS-1991](https://jira.mongodb.org/browse/DRIVERS-1991). Merging.

**Pull request merged**